server/src/lib.rs

```diff
@@ -494,6 +562,23 @@ fn persist_bytes_in_background(data: Bytes, store: Arc<ObjectStore>, location: O
     });
 }
 
+// get bytes from the location in object store
+async fn get_store_bytes(
+    location: &ObjectStorePath,
+    store: &Arc<ObjectStore>,
+) -> Result<bytes::BytesMut> {
+    let b = store
+        .get(&location)
+        .await
+        .context(StoreError)?
+        .map_ok(|b| bytes::BytesMut::from(&b[..]))
+        .try_concat()
+        .await
+        .context(StoreError)?;
+
+    Ok(b)
+}
```

Comment:
I think that this is the first time I think I have seen an `Arc` taken by reference. I thought about it for a moment as it seemed unusual. 

I could imagine that it could make sense to take an `Arc` by reference when the contents of the function isn't sure if it will need to take shared-ownership of the contents of the `Arc` (and then clone it), but in this case it just looks like you might want to call a method on an `&ObjectStore`?

```suggestion
async fn get_store_bytes(
    location: &ObjectStorePath,
    store: &ObjectStore,
) -> Result<bytes::BytesMut> {
    store
        .get(&location)
        .await
        .context(StoreError)?
        .map_ok(|b| bytes::BytesMut::from(&b[..]))
        .try_concat()
        .await
        .context(StoreError)
}
```

Reply:
fixed in b3454daf7097d7f70a535b5ff09b6f75caea3dd0